addons/web/static/src/search/control_panel/control_panel.js

```diff
@@ -233,4 +331,210 @@ export class ControlPanel extends Component {
             ev.stopPropagation();
         }
     }
+
+    /**
+     * @param {TopBarAction} action
+     */
+    _checkValueLocalStorage(action) {
+        const actionIdStr = action.id.toString();
+        return this.state.topbarInfos.visibleTopBarActions[actionIdStr];
+    }
+
+    /**
+     * The selected action is put into (or removed from) the localStorage and its visibility changes.
+     * The state variable visibleTopBarActions keeps track of the visible actions to avoid  having to parse
+     * the localStorage values every time we want to access them.
+     * @param {TopBarAction} action
+     */
+    _setVisibility(action) {
+        const actionIdStr = action.id.toString();
+        if (this.state.topbarInfos.visibleTopBarActions[actionIdStr]) {
+            delete this.state.topbarInfos.visibleTopBarActions[actionIdStr];
+        } else {
+            this.state.topbarInfos.visibleTopBarActions[actionIdStr] = true;
+        }
+        localStorage.setItem(
+            this.localStorageKeyActionIdResId,
+            JSON.stringify(this.state.topbarInfos.visibleTopBarActions)
+        );
+    }
+
+    _onShareCheckboxChange() {
+        this.state.topbarInfos.newActionIsShared = !this.state.topbarInfos.newActionIsShared;
+    }
+
+    /**
+     * @param {Event} ev
+     */
+    async _saveNewAction(ev) {
+        const {
+            newActionName,
+            newActionIsShared,
+            currentTopBarAction,
+            topBarActions,
+            visibleTopBarActions,
+        } = this.state.topbarInfos;
+        if (!newActionName) {
+            this.notificationService.add(_t("A name for your new action is required."), {
+                type: "danger",
+            });
+            ev.stopPropagation();
+            return this.newActionNameRef.el.focus();
+        }
+        const duplicateName = topBarActions.some(({ name }) => name === newActionName);
+        if (duplicateName) {
+            this.notificationService.add(_t("An action with the same name already exists."), {
+                type: "danger",
+            });
+            ev.stopPropagation();
+            return this.newActionNameRef.el.focus();
+        }
+        const userId = newActionIsShared ? false : user.userId;
+        const { action_id, parent_action_id } = currentTopBarAction;
+        const values = {
+            ...currentTopBarAction,
+            parent_action_id: parent_action_id[0],
+            action_id: action_id[0],
```

Comment:
@auon-odoo follow-up to my previous comment that is saying where I added lines to delete the crappy logic in the python model file.

Reply:
Well now that I delete the lines, the response to the comment does not appear :/ the "crappy logic" reference to the logic in the override of the create method in the model `ir.actions.topbar_actions`